lib/collection.go

```diff
@@ -71,7 +72,35 @@ func (collectionImpl) List(scope scope, p *ListParams) ([]dsref.VersionInfo, err
 			Offset:  p.Offset,
 			Limit:   p.Limit,
 		}
-		return s.List(scope.ctx, scope.ActiveProfile().ID, lp)
+		infos, err := s.List(scope.ctx, scope.ActiveProfile().ID, lp)
+		if err != nil {
+			return nil, err
+		}
+
+		if p.EnsureFSIExists {
+			// TODO(b5): adding a defensive check for the existence of the FSI
+			// subsystem in the hopes that cloud code won't follow this path. Haven't
+			// confirmed that's the case
+			if scope.FSISubsystem() != nil {
```

Comment:
AFAIK this has no bearing on cloud using FSI. FSI seems to be set and initialized on every instance creation including cloud. Cloud just simply didn't need/encounter it thus far. We should add some config check on `scope.FSISubsystem()` and return `nil` if configured so to make this viable.